### ISSUE TITLE
new node with raft topology could n't start while all nodes will be not upgraded

### ISSUE THREAD

**Issue by aleksbykov** [scylladb/scylladb#17860]
Base version: 5.4.4-0.20240228.58a1be93b212 with build-id f385c89e3d29e70e0a8beab1dfefcace5e909775
Target version: 5.5.0~dev-0.20240313.15bea069a911 with build-id babc816420ca865129a6fb6d6028c7da05ba478a

Test scenario:
1. Cluster create with base version with version 5.4 
2. Added new node with target version 5.5 and enabled experimental feature `consistent-topology-changes` (directores were created but node is not started yet)
3. then new node is starting and it failed to start:
```
INFO  2024-03-18 17:48:00,834 [shard 0:main] init - Shutting down sighup was successful
INFO  2024-03-18 17:48:00,834 [shard 0:main] init - Shutting down configurables
INFO  2024-03-18 17:48:00,834 [shard 0:main] init - Shutting down configurables was successful
ERROR 2024-03-18 17:48:00,834 [shard 0:main] init - Startup failed: seastar::rpc::unknown_verb_error (unknown verb)
```

**Comment by kbr-scylla** [comment]
@aleksbykov is this an existing scenario in dtests?

Do we need to support it? (Adding a node in new version to cluster in old version)

@avikivity  @mykaul @tzach

**Comment by aleksbykov** [comment]
> is this an existing scenario in dtests?

Yes, we have such test in dtest

**Comment by kbr-scylla** [comment]
@piodul we do the shadow round still on the joining procedure right?
We could potentially skip the new verb join handshake if we see that cluster does not have the consistent topology changes cluster feature...

But to be honest, I don't really see any value in supporting this.

If one wants to join node to 5.4 cluster they can use 5.4 node.

We're just multiplying cases unnecessarily if we want to support this.

> Yes, we have such test in dtest

Which one?

**Comment by piodul** [comment]
> @piodul we do the shadow round still on the joining procedure right? We could potentially skip the new verb join handshake if we see that cluster does not have the consistent topology changes cluster feature...

Yes, but the new verb (`JOIN_NODE_QUERY`) is sent before shadow round starts. Its only purpose is to determine whether we should join in raft topology mode or not. We could handle the "unknown verb" error and decide to fall back to the legacy mode.

> 
> But to be honest, I don't really see any value in supporting this.

The same issue might happen if you try to add a 5.5 node to a cluster that is in the middle of the upgrade (it will happen if the group 0 leader is on 5.4). What about this scenario?

**Comment by kbr-scylla** [comment]
> The same issue might happen if you try to add a 5.5 node to a cluster that is in the middle of the upgrade (it will happen if the group 0 leader is on 5.4). What about this scenario?

We can add 5.4 node instead.

**Comment by kbr-scylla** [comment]
I don't think we will support this, and there is an easy workaround (you want to add node to mixed cluster -- add it in the pre-upgrade version).

Assigning yellowfin.
